parquet/src/main/java/org/apache/iceberg/parquet/VariantReaderBuilder.java

```diff
@@ -198,15 +201,41 @@ public Optional<VariantValueReader> visit(DateLogicalTypeAnnotation ignored) {
 
     @Override
     public Optional<VariantValueReader> visit(TimestampLogicalTypeAnnotation logical) {
-      PhysicalType variantType =
-          logical.isAdjustedToUTC() ? PhysicalType.TIMESTAMPTZ : PhysicalType.TIMESTAMPNTZ;
+      PhysicalType variantType;
+
+      switch (logical.getUnit()) {
```

Comment:
Could this reuse the method from the writer builder? It looks like the same logic, but inline here.